src/openvic-simulation/country/CountryInstance.cpp

```diff
@@ -1552,15 +1625,73 @@ void CountryInstance::update_gamestate(InstanceManager& instance_manager) {
 	}
 }
 
-void CountryInstance::country_reset_before_tick() {
-	for (auto pair : goods_data) {
-		pair.second.clear_daily_recorded_data();
+void CountryInstance::country_tick_before_map(InstanceManager& instance_manager) {
+	//TODO AI sliders
+	//TODO stockpile management
+
+	const fixed_point_t projected_administration_spending = administration_spending_slider_value.get_value() * projected_administration_spending_unscaled_by_slider;
+	const fixed_point_t projected_education_spending = education_spending_slider_value.get_value() * projected_education_spending_unscaled_by_slider;
+	const fixed_point_t projected_military_spending = military_spending_slider_value.get_value() * projected_military_spending_unscaled_by_slider;
+	const fixed_point_t projected_social_spending = social_spending_slider_value.get_value() * (
+		projected_pensions_spending_unscaled_by_slider
+		+ projected_unemployment_subsidies_spending_unscaled_by_slider
+	);
+
+	const fixed_point_t projected_spending = projected_administration_spending
+		+ projected_education_spending
+		+ projected_military_spending
+		+ projected_social_spending;
+	// + tariffs (if negative)
+	// + reparations + war subsidies
+	// + stockpile costs
+	// + industrial subsidies
+	// + loan interest
+
+	fixed_point_t available_funds = cash_stockpile.get_copy_of_value();
+	if (projected_spending <= available_funds) {
+		actual_administration_spending = projected_administration_spending;
+		actual_education_spending = projected_education_spending;
+		actual_military_spending = projected_military_spending;
+		actual_social_spending = projected_social_spending;
+	} else {
+		//TODO try take loan (callback?)
+		//update available_funds with loan
+
+		if (available_funds < projected_education_spending) {
+			actual_education_spending = available_funds;
+			actual_administration_spending = actual_military_spending = actual_social_spending = fixed_point_t::_0();
+		} else {
+			available_funds -= projected_education_spending;
+			actual_education_spending = projected_education_spending;
+
+			if (available_funds < projected_administration_spending) {
+				actual_administration_spending = available_funds;
+				actual_military_spending = actual_social_spending = fixed_point_t::_0();
+			} else {
+				available_funds -= projected_administration_spending;
+				actual_administration_spending = projected_administration_spending;
+
+				if (available_funds < projected_social_spending) {
+					actual_social_spending = available_funds;
+					actual_military_spending = fixed_point_t::_0();
+				} else {
+					available_funds -= projected_social_spending;
+					actual_social_spending = projected_social_spending;
+
+					actual_military_spending = std::min(available_funds, projected_military_spending);
+				}
+			}
+		}
+	}
+
+	for (auto const& [good_instance, data] : goods_data) {
```

Comment:
```suggestion
	for (auto& data : goods_data.get_values()) {
```
You can iterate over an `IndexedMap`'s values directly

Reply:
Cool, let's do that then :)